client/client-lib/src/ln/mod.rs

```diff
@@ -399,9 +399,11 @@ mod tests {
                 .mint
                 .lock()
                 .await
-                .fetch_from_all(|m, db| async {
-                    m.get_contract_account(&mut db.begin_transaction().await, contract)
-                        .await
+                .fetch_from_all(|m, db, module_instance_id| async {
+                    let mut dbtx = db.begin_transaction().await;
+                    let mut module_dbtx =
+                        ModuleDatabaseTransaction::new(&mut dbtx, module_instance_id.clone());
```

Comment:
Can we make the API more like:

```rust
let mut dbtx = db.begin_transaction().await;
let mut module_dbtx = dbtx.module_tx(module_instance_id.clone());
```

Reply:
Oh, I see dpc commented the same in the thread :laughing:

Reply:
Yeah I tried this initially and was running into lifetime problems. I'll try and do the refactoring @dpc suggested and ping if I'm still running into issues